src/iinit.cpp

```diff
@@ -19,8 +19,32 @@
 
 #include <nlohmann/json.hpp>
 
-#include <iostream>
+#include <algorithm>
 #include <fstream>
+#include <iostream>
+
+namespace
+{
+    namespace irods_auth = irods::experimental::auth;
+
+    constexpr const char* const AUTH_OPENID_SCHEME = "openid";
+    constexpr const char* const PAM_INTERACTIVE_SCHEME = "pam_interactive";
+
+    auto scheme_uses_iinit_password_prompt(const std::string_view _scheme) -> bool
+    {
+        const std::initializer_list<const std::string_view> no_password_prompt = {
+            AUTH_OPENID_SCHEME,
+            irods::AUTH_GSI_SCHEME,
+            irods::AUTH_PAM_SCHEME,
+            irods_auth::scheme::pam_password,
+            PAM_INTERACTIVE_SCHEME
```

Comment:
remind me again why we have two different flavors of magic string here?

can we make `irods_auth::scheme::pam_password` also... the same as the others?   or change all the others?

Reply:
This is kind of a weird spot in which we find ourselves...

Historically, the iRODS core code (or, I guess in this case it would be the client libraries) has maintained the scheme names of the auth plugins. As you can see, some of these are actually maintained by `iinit` itself.

In the new auth plugin framework work, I thought it might be a good idea for each plugin to provide its own scheme name, which is why `pam_password` appears the way that it is. Looking back on this, maybe this is not such a good idea because any external plugins which may or may not be installed cannot be supported like this.

That being said, I think this problem is specific to `iinit` and the plugins will not need to provide their own scheme names in our next authentication client.

So... I'm willing to drop the `irods_auth::scheme::pam_password` in favor of a local string if that seems better.

Reply:
I defer.   Consistency is good - but also, this is all going to change before too long...

Reply:
This is definitely the grown-up solution.    Now... how do we get there gracefully...

Reply:
I've created https://github.com/irods/irods/issues/6752 to investigate this .

For the near term, I think that a file-scoped constant would be good to replace this silly thing. :)

Reply:
Done

Reply:
> For the near term, I think that a file-scoped constant would be good to replace this silly thing. :)

I agree.